app/database/models/ms_schema/tasks_list.py

```diff
@@ -0,0 +1,210 @@
+        }
+        self.next_task_id += 1
+        self.tasks = self.tasks + [task]
+
+    def delete_task(self, task_id: int) -> None:
+        """Deletes a task from the list of tasks.
+
+        Args:
+            task_id: Id of the task to be deleted.
+        """
+
+        new_list = list(
+            filter(lambda task: task[TasksFields.ID.value] != task_id, self.tasks)
+        )
+
+        self.tasks = new_list
+        self.save_to_db()
+
+    def update_task(
+        self,
+        task_id: int,
+        description: str = None,
+        is_done: bool = None,
+        completed_at: date = None,
+    ) -> None:
+        """Updates a task.
+        
+        Args:
+            task_id: Id of the task to be updated.
+            description: A description of the task.
+            created_at: Date on which the task is created.
+            is_done: Boolean specifying completion of the task.
+            completed_at: Date on which task is completed.
+        """
+
+        new_list = []
+        for task in self.tasks:
+            if task[TasksFields.ID.value] == task_id:
+                new_task = task.copy()
+                if description is not None:
+                    new_task[TasksFields.DESCRIPTION.value] = description
+
+                if is_done is not None:
+                    new_task[TasksFields.IS_DONE.value] = is_done
+
+                if completed_at is not None:
+                    new_task[TasksFields.COMPLETED_AT.value] = completed_at
+
+                new_list = new_list + [new_task]
+                continue
+
+            new_list = new_list + [task]
+
+        self.tasks = new_list
+        self.save_to_db()
+
+    def find_task_by_id(self, task_id: int):
+        """Returns the task that has the specified id.
+        
+        Args:
+            task_id: Id of the task.
+
+        Returns:
+            The task instance.
+        """
+        task = list(
+            filter(lambda task: task[TasksFields.ID.value] == task_id, self.tasks)
+        )
+        if len(task) == 0:
+            return None
+        return task[0]
+
+    def is_empty(self) -> bool:
+        """Checks if the list of tasks is empty.
+
+        Returns:
+            Boolean; True if the task is empty, False otherwise.
+        """
+
+        return len(self.tasks) == 0
+
+    def json(self):
+        """Creates json object of the attributes of list of tasks.
+
+        Returns:
+            Json objects of attributes of list of tasks.
+        """
+
+        return {
+            "id": self.id,
+            "mentorship_relation_id": self.mentorship_relation_id,
+            "tasks": self.tasks,
+            "next_task_id": self.next_task_id,
+        }
+
+    def __repr__(self):
+        """Creates a representation of an object.
+        
+        Returns:
+            A string representation of the task object.
+        """
+
+        return "Task | id = %s; tasks = %s; next task id = %s" % (
```

Comment:
Use f strings for formatting

Reply:
I agree... been always consistent on the BIT code I wrote my self. didn't check the MS code I copied. will change it for BIT